contracts/Pool.sol

```diff
@@ -1,31 +1,292 @@
+    /// On every block this amount is rounded down to the closest multiple of the sum of the weights
+    /// of the receivers and split between all sender's receivers proportionally to their weights.
+    /// Each receiver receives then their part from the sender's balance.
+    /// If set to zero, stops sending any funds.
+    /// @param amount The target per-block amount
+    function setAmountPerBlock(uint256 amount) public pauseSending {
+        senders[msg.sender].amtPerBlock = uint192(amount);
+    }
+
+    /// @notice Sets the weight of a receiver of the sender of the message.
+    /// The weight regulates the share of the amount being sent on every block in relation to
+    /// other sender's receivers.
+    /// Setting a non-zero weight for a new receiver, added it to the list of sender's receivers.
+    /// Setting the zero weight for a receiver, removes it from the list of sender's receivers.
+    /// @param receiver The address of the receiver
+    /// @param weight The weight of the receiver
+    function setReceiver(address receiver, uint32 weight) public pauseSending {
+        Sender storage sender = senders[msg.sender];
+        uint32 oldWeight = sender.receiverWeights.setWeight(receiver, weight);
+        sender.weightSum -= oldWeight;
+        sender.weightSum += weight;
+        require(
+            sender.weightSum <= SENDER_WEIGHTS_SUM_MAX,
+            "Too much total receivers weight"
+        );
+        if (weight != 0 && oldWeight == 0) {
+            sender.weightCount++;
+            require(
+                sender.weightSum <= SENDER_WEIGHTS_COUNT_MAX,
+                "Too many receivers"
+            );
+        } else if (weight == 0 && oldWeight != 0) {
+            sender.weightCount--;
+        }
+    }
+
+    /// @notice Pauses all sender's payments on current block allowing safe configuration
+    modifier pauseSending {
+        stopSending();
+        _;
+        startSending();
+    }
+
+    /// @notice Stops all sender's payments on current block
+    function stopSending() internal {
+        uint64 blockNumber = uint64(block.number);
+        Sender storage sender = senders[msg.sender];
+        // Hasn't been sending anything
+        if (sender.weightSum == 0 || sender.amtPerBlock < sender.weightSum)
+            return;
+        uint192 amtPerWeight = sender.amtPerBlock / sender.weightSum;
+        uint192 amtPerBlock = amtPerWeight * sender.weightSum;
+        uint256 endBlockUncapped = sender.startBlock +
+            uint256(sender.startBalance / amtPerBlock);
+        uint64 endBlock = endBlockUncapped > maxBlockNumber
+            ? maxBlockNumber
+            : uint64(endBlockUncapped);
+        // All donations have run out
+        if (endBlock <= blockNumber) {
+            sender.startBalance %= amtPerBlock;
+            return;
+        }
+        sender.startBalance -= (blockNumber - sender.startBlock) * amtPerBlock;
+        setDeltasFromNow(-int256(amtPerWeight), endBlock);
```

Comment:
We can't, if we don't call `setDeltasFromNow`, we won't remove effects of the sender from all its receivers' deltas describing the future. This would mean that if we remove one of the receivers between `stopSending` and `startSending`, it would keep receiving funds, just from thin air.